vllm/entrypoints/openai/serving_chat.py

```diff
@@ -1208,13 +1208,28 @@ async def chat_completion_stream_generator(
                             # check to see if there's anything left to stream
                             remaining_call = expected_call.replace(actual_call, "", 1)
                             # set that as a delta message
+                            # preserve id/type/name from original delta_message
+                            original_tc = next(
+                                (
+                                    tc
+                                    for tc in delta_message.tool_calls
+                                    if tc.index == index
+                                ),
+                                None,
+                            )
+                            original_fn = original_tc.function if original_tc else None
```

Comment:
Thanks~ @amittell 

I noticed that #31451 seems to be addressing the same issue.

What do you think about #31451?